plugin/core/transports.py

```diff
@@ -176,27 +176,7 @@ def _stderr_loop(self) -> None:
         self._send_queue.put_nowait(None)
 
 
-def create_transport(config: ClientConfig, cwd: Optional[str], window: sublime.Window,
-                     callback_object: TransportCallbacks, variables: Dict[str, str]) -> JsonRpcTransport:
-    tcp_port = None  # type: Optional[int]
-    if config.tcp_port is not None:
-        tcp_port = _find_free_port() if config.tcp_port == 0 else config.tcp_port
-    if tcp_port is not None:
-        variables["port"] = str(tcp_port)
-    args = sublime.expand_variables(config.command, variables)
-    args = [os.path.expanduser(arg) for arg in args]
-    if tcp_port is not None:
-        # DEPRECATED -- replace {port} with $port or ${port} in your client config
-        args = [a.replace('{port}', str(tcp_port)) for a in args]
-    env = os.environ.copy()
-    for var, value in config.env.items():
-        env[var] = sublime.expand_variables(value, variables)
-    if tcp_port is not None:
-        stdout = subprocess.DEVNULL
-        stdin = subprocess.DEVNULL
-    else:
-        stdout = subprocess.PIPE
-        stdin = subprocess.PIPE
+def _fixup_startup_args(args: List[str]) -> Any:
```

Comment:
Can we sort the function in the order those are used (roughly at least)? Or are you intentionally putting private function on top?

Reply:
Do the last three commits improve the ordering?

Reply:
Yes. And diff is easier to read also.